src/tests/performance/blocks.rs

```diff
@@ -0,0 +1,222 @@
+use comfy_table::{modifiers::UTF8_ROUND_CORNERS, presets::UTF8_FULL, Cell, CellAlignment, Table};
+use histogram::Histogram;
+use std::{collections::VecDeque, convert::TryFrom};
+use tokio::time::{timeout, Duration};
+
+use crate::{
+    helpers::initiate_handshake,
+    protocol::{
+        message::{Message, MessageFilter},
+        payload::{block::Block, Inv},
+    },
+    setup::{
+        config::new_local_addr,
+        node::{Action, Node},
+    },
+};
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+async fn getdata_blocks_latency() {
+    // ZG-PERFORMANCE-001, GetData-Block latency
+    //
+    // The node behaves as expected under load from other peers.
+    //
+    // We test the overall performance of a node's GetData-Block latency.
+    //
+    // Note: This test does not assert any requirements, but requires manual inspection
+    //       of the results table. This is because the results will rely on the machine
+    //       running the test.
+    //
+    // ZCashd: Strange, small slow-down as soon as multiple peers are present.
+    //         Otherwise performs consistently up to around 200 concurrent peers.
+    //
+    // Zebra: Chokes right out of the gate, likely to the DoS behaviour of spamming GetAddr and GetData.
```

Comment:
Good find 👍 
This might be indicative of throttling, since we're making the same requests from the same addresses?

Reply:
It's possible, but I couldn't really make sense of what's happening.

It would really help if the logs could be sent to an elastic search type database. The zebra logs are structured (I think), so that would let us filter them. But zcashd looks like vanilla logs, so wouldn't help too much there.